src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt

```diff
@@ -117,47 +117,27 @@ class LicenceConditionService(
       .orElseThrow { EntityNotFoundException("$licenceId") }
 
     val username = SecurityContextHolder.getContext().authentication.name
-    val submittedAdditionalConditions =
+    val submittedConditions =
       request.additionalConditions.transformToEntityAdditional(licenceEntity, request.conditionType)
 
-    val newAdditionalConditions =
-      submittedAdditionalConditions.filter { licenceEntity.additionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode } }
-        .map { newAdditionalCondition ->
-          newAdditionalCondition.copy(
-            expandedConditionText = newAdditionalCondition.conditionText,
-            licence = licenceEntity,
-          )
-        }
+    val existingConditions = licenceEntity.additionalConditions
 
-    val removedAdditionalConditionsList =
-      licenceEntity.additionalConditions.filter {
-        it.conditionType == request.conditionType && submittedAdditionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode }
-      }
+    val newConditions = existingConditions.getAddedConditions(submittedConditions)
 
-    val updatedAdditionalConditions =
-      licenceEntity.additionalConditions.filter { condition -> removedAdditionalConditionsList.none { it.conditionCode == condition.conditionCode } }
+    val removedConditions = existingConditions.getRemovedConditions(submittedConditions, request)
 
-    submittedAdditionalConditions.forEach { newAdditionalCondition ->
-      updatedAdditionalConditions.filter {
-        it.conditionCode == newAdditionalCondition.conditionCode
-      }.forEach {
-        it.conditionCategory = newAdditionalCondition.conditionCategory
-        it.conditionText = newAdditionalCondition.conditionText
-        it.conditionSequence = newAdditionalCondition.conditionSequence
-        it.conditionType = newAdditionalCondition.conditionType
```

Comment:
not sure why this doesn't update the expanded text too!

Reply:
I think this updating for expanded text is done in the `updateAdditionalConditionData` function in the same service. No idea why?